src/java.desktop/share/classes/javax/swing/DefaultListSelectionModel.java

```diff
@@ -686,23 +690,27 @@ public void insertIndexInterval(int index, int length, boolean before)
      * the selection model.  This is typically called to sync the selection
      * model width a corresponding change in the data model.  Note
      * that (as always) index0 need not be &lt;= index1.
+     *
+     * @throws IndexOutOfBoundsException if either index is negative
+     *         or equal to Integer.MAX_VALUE
      */
     public void removeIndexInterval(int index0, int index1)
     {
+        if (index0 < 0 || index1 < 0) {
+            throw new IndexOutOfBoundsException("index is negative");
+        }
+        if (index0 == Integer.MAX_VALUE || index1 == Integer.MAX_VALUE) {
+            throw new IndexOutOfBoundsException("index is equal to MAX_VALUE");
+        }
         int rmMinIndex = Math.min(index0, index1);
         int rmMaxIndex = Math.max(index0, index1);
-        int gapLength = ((rmMaxIndex - rmMinIndex) + 1) > (rmMaxIndex - rmMinIndex)
-                ? ((rmMaxIndex - rmMinIndex) + 1) : (rmMaxIndex - rmMinIndex);
+        int gapLength = (rmMaxIndex - rmMinIndex) + 1;
 
         /* Shift the entire bitset to the left to close the index0, index1
          * gap.
          */
         for(int i = rmMinIndex; i <= maxIndex; i++) {
-            if ((i + gapLength) > gapLength) {
-                setState(i, value.get(i + gapLength));
-            } else {
-                setState(i, value.get(gapLength));
-            }
+            setState(i, value.get(i + gapLength));
         }
```

Comment:
The `for`-loop should look like this:
```java
        for(int i = rmMinIndex; i >= 0 && i <= maxIndex; i++) {
            setState(i, (i <= Integer.MAX_VALUE - gapLength)
                        && (i + gapLength >= minIndex)
                        && value.get(i + gapLength));
        }
```
It breaks if `i` becomes negative. In the body, the first condition `(i <= Integer.MAX_VALUE - gapLength)` prevents accessing indexes greater than `Integer.MAX_VALUE`, the second condition `(i + gapLength >= minIndex)` is an optimisation, the values for indexes below `minIndex` are `false`, so the call to `value.get` is skipped. (The second condition is not required.)